content/news/2023-03-04-bevy-0.10/index.md

```diff
@@ -0,0 +1,524 @@
+        *next_state = NextState(AppState::InGame);
+    }
+}
+```
+
+But wait you say: what about my state stack? My elaborate queued transitions?! My meticulous error handling on every operation that I definitely didn't just unwrap?!!
+
+In practice, we found that the state stack was a) very complex to learn b) very prone to exasperating bugs c) mostly ignored.
+As a result, states are now "stackless": only one queued state of each type at a time.
+
+Thanks to the help of some brave alpha testers, we're reasonably confident that this shouldn't be too bad to migrate away from.
+If you were relying on the state stack, you might choose to:
+
+* rearchitect some of that logic out of states
+* use additional state types, which capture orthogonal elements of your app's status
+* build your own state stack abstraction using the same patterns as Bevy's first-party version: please let the rest of the community know so you can collaborate!
+
+## Base Sets: Getting Default Behavior Right
+
+Of course the skeptical reader may point out that:
+
+1. Bevy automatically runs its systems in parallel.
+2. [The order of systems is nondeterministic unless there is an explicit ordering relationship between them](https://github.com/bevyengine/bevy/blob/latest/examples/ecs/nondeterministic_system_order.rs)?
+3. All of the systems are now stored in a single `Schedule` object with no barriers between them?
+4. Systems can belong to any number of system sets, each of which can add their own behavior?
+
+Won't this lead to utter chaos and tedious spaghetti-flavored work to resolve every last ordering ambiguity?
+Many users _liked_ stages, they were helpful for understanding the structure of my app!
+
+Well, I'm glad you asked, rhetorical skeptic. To reduce this chaos (and ease migration), Bevy 0.10 comes with a brand new collection of system sets with the default plugins: [`CoreSet`](https://dev-docs.bevyengine.org/bevy/app/enum.CoreSet.html), [`StartupSet`](https://dev-docs.bevyengine.org/bevy/app/enum.StartupSet.html) and [`RenderSet`](https://dev-docs.bevyengine.org/bevy/render/enum.RenderSet.html). The similarity of their names to [`CoreStage`](https://docs.rs/bevy/0.9.1/bevy/app/enum.CoreStage.html), [`StartupStage`](https://docs.rs/bevy/0.9.1/bevy/app/enum.StartupStage.html) and [`RenderStage`](https://docs.rs/bevy/0.9.1/bevy/render/enum.RenderStage.html) is not a coincidence: there are command flush points between each set, and existing systems have been migrated directly.
+
+Some parts of the stage-centric architecture were appealing: a clear high level structure, coordination on flush points (to reduce excessive bottlenecks) and good default behavior.
+To keep those bits (while excising the frustrating ones), we've introduced the concept of **base sets**, added in [#7466](https://github.com/bevyengine/bevy/pull/7466) by `@cart`. Base sets are system sets, except:
+
+1. Every system (but not every system set) must belong to exactly one base set.
```

Comment:
I thought that systems can have no base set:
```
.add_system(
    no_base_set
        .no_default_base_set(),
)
```

Reply:
Yup this is true. Schedules also don't have a "default base set" unless one is set, and this is still valid. We should rephrase this.